design/use-statement.md

```diff
@@ -76,22 +80,35 @@ resources:
 
 steps:
 - use: someTool someVersion
-  auth: myFeed
+  authTo: myFeed
 ```
 
-`feed` is not yet a resource type, but it's coming.
+(`feed` is not yet a resource type, but it's coming.)
+
+Tasks which implement this contract can expect a service connection at runtime.
+It's up to the resource provider to convert other forms of auth (such as a raw username + password) into a service connection.
+
+Some ecosystems will require optional, additional inputs.
+Just like on `task`, you can pass a map of `inputs`.
+```yaml
+- use: python
+  inputs:
+    architecture: x64
+```
 
-No additional keywords are reserved.
-Any additional keywords are passed through as `inputs` to the underlying task.
+We need to allow tasks the ability to remove fields in major versions.
+Therefore, if a task doesn't accept one of the inputs, we won't fail the build but we will inject a warning about the mismatch.
+Depending on the scenario, the build could still fail at runtime.
+This makes the issue debuggable even if not ideal.
 
 ## Schema
 
 ```yaml
-- use: string       # required tool name and optional version
-  version: string   # optional version (specifying version twice is an error)
-  proxy: boolean    # whether to install proxy information; defaults to true
-  auth: string      # optional name of a resource to derive auth information from
-  {string: any}     # additional arguments to pass as task inputs
+- use: string            # required tool name
+  version: string        # optional version
+  proxy: boolean         # whether to install proxy information; defaults to true
+  authTo: string         # optional name of a resource to derive auth information from
```

Comment:
Can this be `string | [ string ]`? There are cases where we'll want to provide multiple authenticated resources.

Reply:
Sure, that makes sense. The contract will be that tasks have to handle an array, even if it's cardinality 1. We'll need to have a convention for ecosystems that can't handle multiple auth, such as npm, right?

Reply:
True, so probably a bad example. I assume the npm service connection brings along the registry details, so the task could do the right thing if there are multiple.